### PR TITLE
Joint roach fix

### PR DESCRIPTION
**Pull request by Cyrano7**
From issue https://github.com/CleverRaven/Cataclysm-DDA/issues/15931 Moves joint lit and joint roach to generic.
Edit: And changes joint roach to generic.

### CODE REVIEW

**Comment by illi-kun** [comment]
It's not an actual fix but it's useful change anyway.

**Comment by Cyrano7** [comment]
Wont this make the joint roach not edible?

**Comment by illi-kun** [comment]
Oh, I'm sorry I've missed that part.

**Comment by Cyrano7** [comment]
ya I should update the description its not really clear.

**Comment by illi-kun** [comment]
Please change the description of PR using ["magic words"](https://help.github.com/articles/closing-issues-via-commit-messages/), addition of keywords "fixes #15931" will automatically close the referenced issue when your PR will be merged.

**Comment by mugling** [comment]
@Cyrano7 I've appended the example wording to your description

File: data/json/items/generic.json [hunk 1]
```diff
@@ -2451,6 +2451,35 @@
...
     },
     {
+        "type" : "GENERIC",
+        "id" : "joint_lit",
+        "name" : "joint",
+        "weight" : 0,
+        "volume" : 0,
+        "color" : "ltgreen",
+        "symbol" : "!",
+        "to_hit" : -2,
```
Comment (mugling) [inline data/json/items/generic.json]: Doesn't need `to_hit`, `cutting` etc

File: data/json/items/generic.json [hunk 2]
```diff
@@ -2451,6 +2451,35 @@
         "category": "other"
     },
     {
+        "type" : "GENERIC",
+        "id" : "joint_lit",
+        "name" : "joint",
+        "weight" : 0,
```
Comment (mugling) [inline data/json/items/generic.json]: `weight` and `volume` are also probably redundant

**Comment by Cyrano7** [comment]
@mugling ahhhhh silly me, that's makes more sense.

**Comment by Cyrano7** [comment]
@mugling Thanks!

**Comment by Rivet-the-Zombie** [comment]
What about smoking them? It's a fairly normal thing IRL to smoke roaches.

**Comment by mugling** [comment]
Are they not craftable into new joints (like cigarette stubs?)

**Comment by karols146** [comment]
They should work same like cigarettes, You can craft normal joint with 5 roaches if i remember right.

**Comment by Rivet-the-Zombie** [comment]
They are, but sometimes you might just want to burn the roach on its own.

**Comment by karols146** [comment]
So if we going that way, you should can burn butts too

**Comment by Cyrano7** [comment]
and cigar butts too.

**Comment by illi-kun** [comment]
I don't think that general idea of that PR is have to be implemented. What it adds to the game? Previously, it was clear path from something useful (cigarette) to something almost useless (cigarette butt, essentially a trash but can be utilized in something useful if you ready to collect some trash for a minor benefit).

As far as I understood your PR, now an extra step is added to this consumption chain, so now it's cig -> cig butt (-> cig butt butt??). Why it's useful? Why the consumer doesn't smoke it to the unsmokable condition at first attempt? What it brings to the game except excessive keypresses and hidden mechanic (not all players will even try to smoke it)?

**Pull request merged**